### DISCUSSION TITLE
Crash if using a poor template syntax for tasks arguments

### DISCUSSION THREAD

**Discussion by rik** [jdx/mise#3879] [General]
In a TOML task definition, forgetting quotes around the name in an argument template will crash mise.

```
[tasks.foo]
run = ['foo {{arg(name=jobs)}}']
```

```
$ mise run foo
The application panicked (crashed).
Message:  called `Result::unwrap()` on an `Err` value: Error { kind: Msg("Failed to render '__tera_one_off'"), source: Some(Error { kind: CallFunction("arg"), source: Some(Error { kind: Msg("Variable `jobs` not found in context while rendering '__tera_one_off'"), source: None }) }) }
Location: src/task/task_script_parser.rs:284

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

It also crashes when running `mise tasks info foo`

**Comment by Marukome0743** [comment]
Thanks for reporting this. Invalid task templates are now handled as regular errors instead of causing mise to panic. The fix was introduced in [#3904](https://github.com/jdx/mise/pull/3904) and released in [mise v2025.1.0](https://github.com/jdx/mise/releases/tag/v2025.1.0).

I verified the original example against the current `main` branch:

```toml
[tasks.foo]
run = ['foo {{arg(name=jobs)}}']
```

`mise run foo` now exits with a normal template-rendering error explaining that the `name` argument is invalid, and `mise tasks info foo` no longer crashes either.

The argument name should be quoted, for example:

```toml
run = ['foo {{arg(name="jobs")}}']
```

Upgrading to a recent mise release should resolve the reported panic.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
